xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-test/xwiki-platform-flamingo-skin-test-docker/src/test/it/org/xwiki/flamingo/test/docker/DeletePageIT.java

```diff
@@ -374,4 +380,54 @@ public void deletePageWithUsedClass(TestUtils setup, TestInfo info)
         ViewPage viewPage = setup.gotoPage(freeReference);
         assertFalse(viewPage.exists());
     }
+
+    /**
+     * This test checks the behaviour of the page deletion form when the user is allowed to choose between sending the
+     * page to the recycle bin or delete it permanently.
+     * In this test the user chooses to send the page to the recycle bin.
+     */
+    @Test
+    @Order(9)
+    public void deleteToRecycleBin(TestUtils setup)
+    {
+        // Set the user type to Advanced
+        Map<String, Object> userProperties = new HashMap<>();
+        userProperties.put("usertype", "Advanced");
+        setup.updateObject("XWiki", "superadmin", "XWiki.XWikiUsers", 0, userProperties);
+
+        // Set the canSkipRecycleBin property to true
+        setup.updateObject(REFACTORING_CONFIGURATION_REFERENCE, "Refactoring.Code.RefactoringConfigurationClass", 0,
+            "canSkipRecycleBin", "1");
+
+        setup.gotoPage(SPACE_VALUE, PAGE_VALUE);
+        DeletePageConfirmationPage confirmationPage = this.viewPage.deletePage();
+        confirmationPage.selectOptionToRecycleBin();
+        confirmationPage.clickYes();
+        DeletingPage deletingPage = new DeletingPage();
+        deletingPage.waitUntilFinished();
+        assertEquals(DELETE_SUCCESSFUL, deletingPage.getInfoMessage());
+        DeletePageOutcomePage deleteOutcome = deletingPage.getDeletePageOutcomePage();
+        assertEquals(LOGGED_USERNAME, deleteOutcome.getPageDeleter());
+        assertEquals(DOCUMENT_NOT_FOUND, deleteOutcome.getMessage());
+    }
+
+    /**
+     * This test checks the behaviour of the page deletion form when the user is allowed to choose between sending the
+     * page to the recycle bin or delete it permanently.
+     * In this test the user chooses to delete the page permanently.
+     */
```

Comment:
Do we have tests (unit tests maybe?) when the user is NOT allowed to choose?

Reply:
The UI tests a here https://github.com/xwiki/xwiki-platform/pull/1401/files#diff-2abde92d350206b28dbfc6499abfabc8R372
I have added a step in the docker test to verify that the form is not displayed when the user is advanced but  `isRecycleBinSkippingActivated` is false. (see https://github.com/xwiki/xwiki-platform/pull/1401/files#diff-57f6b731bd337026fb3a5409c237863cR400)